File: android/app/src/main/java/com/purchasely/shaker/ui/screen/settings/SettingsScreen.kt
package com.purchasely.shaker.ui.screen.settings

import android.app.Activity
import android.content.ClipData
import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.ContentCopy
import androidx.compose.material.icons.filled.Logout
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.platform.ClipEntry
import androidx.compose.ui.platform.LocalClipboard
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.purchasely.shaker.data.PurchaselySdkMode
import com.purchasely.shaker.domain.model.DisplayMode
import com.purchasely.shaker.domain.model.ThemeMode
import com.purchasely.shaker.purchasely.DisplayResult
import com.purchasely.shaker.purchasely.PurchaselyWrapper
import com.purchasely.shaker.ui.theme.Shaker
import kotlinx.coroutines.launch
import org.koin.androidx.compose.koinViewModel
import org.koin.compose.koinInject

@Composable
fun SettingsScreen(viewModel: SettingsViewModel = koinViewModel()) {
    val tokens = Shaker.tokens
    val userId by viewModel.userId.collectAsStateWithLifecycle()
    val isPremium by viewModel.isPremium.collectAsStateWithLifecycle()
    val restoreMessage by viewModel.restoreMessage.collectAsStateWithLifecycle()
    val themeMode by viewModel.themeMode.collectAsStateWithLifecycle()
    val sdkMode by viewModel.sdkMode.collectAsStateWithLifecycle()
    val displayMode by viewModel.displayMode.collectAsStateWithLifecycle()
    val anonymousId by viewModel.anonymousId.collectAsStateWithLifecycle()

    val analyticsConsent by viewModel.analyticsConsent.collectAsStateWithLifecycle()
    val identifiedAnalyticsConsent by viewModel.identifiedAnalyticsConsent.collectAsStateWithLifecycle()
    val personalizationConsent by viewModel.personalizationConsent.collectAsStateWithLifecycle()
    val campaignsConsent by viewModel.campaignsConsent.collectAsStateWithLifecycle()
    val thirdPartyConsent by viewModel.thirdPartyConsent.collectAsStateWithLifecycle()

    val clipboard = LocalClipboard.current
    val clipboardScope = rememberCoroutineScope()
    val context = LocalContext.current
    val purchaselyWrapper: PurchaselyWrapper = koinInject()
    var loginInput by remember { mutableStateOf("") }

    LaunchedEffect(restoreMessage) {
        restoreMessage?.let {
            Toast.makeText(context, it, Toast.LENGTH_SHORT).show()
            viewModel.clearRestoreMessage()
        }
    }

    LaunchedEffect(Unit) {
        viewModel.requestPaywallDisplay.collect { handle ->
            val activity = context as? Activity ?: return@collect
            val result = purchaselyWrapper.display(handle, activity)
            when (result) {
                is DisplayResult.Purchased, is DisplayResult.Restored -> viewModel.onPurchaseCompleted()
                else -> {}
            }
        }
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(tokens.bg)
            .statusBarsPadding()
            .verticalScroll(rememberScrollState()),
    ) {
        Text(
            "Settings",
            color = tokens.text,
            fontSize = 28.sp,
            fontWeight = FontWeight.Bold,
            modifier = Modifier.padding(horizontal = 20.dp, vertical = 16.dp),
        )

        SectionHeader("Account")
        SettingsCard {
            if (userId != null) {
                Row(
                    modifier = Modifier.fillMaxWidth().padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Box(
                        modifier = Modifier
                            .size(44.dp)
                            .clip(CircleShape)
                            .background(tokens.indigo),
                        contentAlignment = Alignment.Center,
                    ) {
                        Text(
                            (userId ?: "").take(2).uppercase(),
                            color = tokens.onIndigo,
                            fontWeight = FontWeight.Bold,
                            fontSize = 16.sp,
                        )
                    }
                    Spacer(Modifier.size(12.dp))
                    Column(Modifier.weight(1f)) {
                        Text("Logged in as", color = tokens.textSec, fontSize = 12.sp)
                        Text(userId ?: "", color = tokens.text, fontSize = 16.sp, fontWeight = FontWeight.SemiBold)
                    }
                    Row(
                        modifier = Modifier.clickable { viewModel.logout() },
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Icon(Icons.Default.Logout, null, tint = tokens.danger, modifier = Modifier.size(16.dp))
                        Spacer(Modifier.size(6.dp))
                        Text("Logout", color = tokens.danger, fontSize = 14.sp, fontWeight = FontWeight.SemiBold)
                    }
                }
            } else {
                Row(
                    modifier = Modifier.fillMaxWidth().padding(12.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Box(
                        modifier = Modifier
                            .weight(1f)
                            .height(44.dp)
                            .clip(RoundedCornerShape(12.dp))
                            .background(tokens.bgSubtle)
                            .border(1.dp, tokens.hair, RoundedCornerShape(12.dp))
                            .padding(horizontal = 14.dp),
                        contentAlignment = Alignment.CenterStart,
                    ) {
                        if (loginInput.isEmpty()) {
                            Text("User ID", color = tokens.textSec, fontSize = 15.sp)
                        }
                        BasicTextField(
                            value = loginInput,
                            onValueChange = { loginInput = it },
                            singleLine = true,
                            textStyle = TextStyle(color = tokens.text, fontSize = 15.sp),
                            cursorBrush = SolidColor(tokens.indigo),
                            modifier = Modifier.fillMaxWidth(),
                        )
                    }
                    Spacer(Modifier.size(10.dp))
                    val enabled = loginInput.isNotBlank()
                    Box(
                        modifier = Modifier
                            .height(44.dp)
                            .clip(RoundedCornerShape(12.dp))
                            .background(if (enabled) tokens.accent else tokens.bgSubtle)
                            .clickable(enabled = enabled) {
                                viewModel.login(loginInput)
                                loginInput = ""
                            }
                            .padding(horizontal = 18.dp),
                        contentAlignment = Alignment.Center,
                    ) {
                        Text(
                            "Login",
                            color = if (enabled) Color.White else tokens.textSec,
                            fontWeight = FontWeight.SemiBold,
                            fontSize = 14.sp,
                        )
                    }
                }
            }
            Divider()
            RowBetween {
                Text("Premium status", color = tokens.text, fontSize = 15.sp)
                Box(
                    modifier = Modifier
                        .clip(RoundedCornerShape(100.dp))
                        .background(if (isPremium) tokens.goldSoft else tokens.indigoSoft)
                        .padding(horizontal = 10.dp, vertical = 3.dp),
                ) {
                    Text(
                        if (isPremium) "PRO" else "FREE",
                        color = tokens.indigoText,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Bold,
                    )
                }
            }
            Divider()
            Row(
                modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 12.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Column(Modifier.weight(1f)) {
                    Text("Anonymous ID", color = tokens.textSec, fontSize = 13.sp)
                    Text(anonymousId, color = tokens.text, fontSize = 11.sp, maxLines = 1)
                }
                Box(
                    modifier = Modifier
                        .size(32.dp)
                        .clip(RoundedCornerShape(8.dp))
                        .background(tokens.inputBg)
                        .clickable {
                            clipboardScope.launch {
                                clipboard.setClipEntry(ClipEntry(ClipData.newPlainText("anonymousId", anonymousId)))
                            }
                            Toast.makeText(context, "Copied", Toast.LENGTH_SHORT).show()
                        },
                    contentAlignment = Alignment.Center,
                ) {
                    Icon(Icons.Default.ContentCopy, null, tint = tokens.textSec, modifier = Modifier.size(16.dp))
                }
            }
        }

        SectionHeader("Purchases")
        Column(modifier = Modifier.padding(horizontal = 20.dp), verticalArrangement = Arrangement.spacedBy(10.dp)) {
            OutlineButton(label = "Restore purchases") { viewModel.restorePurchases() }
            OutlineButton(label = "Show onboarding") { viewModel.showOnboardingPaywall() }
        }
        Spacer(Modifier.height(16.dp))

        SectionHeader("Purchasely SDK")
        Segmented(
            options = listOf(PurchaselySdkMode.PAYWALL_OBSERVER.label, PurchaselySdkMode.FULL.label),
            active = sdkMode.label,
            onSelect = { label ->
                val mode = PurchaselySdkMode.entries.firstOrNull { it.label == label } ?: PurchaselySdkMode.DEFAULT
                viewModel.setSdkMode(mode)
            },
        )
        Text(
            "Default mode is Paywall Observer — Shaker observes purchases but uses its own paywall UI.",
            color = tokens.textSec,
            fontSize = 12.sp,
            modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp),
        )
        Spacer(Modifier.height(16.dp))

        SectionHeader("Data privacy")
        SettingsCard {
            ToggleRow("Analytics", "Anonymous audience measurement", analyticsConsent) { viewModel.setAnalyticsConsent(it) }
            Divider()
            ToggleRow("Identified analytics", "User-identified analytics", identifiedAnalyticsConsent) { viewModel.setIdentifiedAnalyticsConsent(it) }
            Divider()
            ToggleRow("Personalization", "Personalized content & offers", personalizationConsent) { viewModel.setPersonalizationConsent(it) }
            Divider()
            ToggleRow("Campaigns", "Promotional campaigns", campaignsConsent) { viewModel.setCampaignsConsent(it) }
            Divider()
            ToggleRow("Third-party integrations", "External analytics & integrations", thirdPartyConsent, last = true) { viewModel.setThirdPartyConsent(it) }
        }
        Text(
            "Technical processing required for app operation cannot be disabled.",
            color = tokens.textSec,
            fontSize = 12.sp,
            modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp),
        )

        SectionHeader("Appearance")
        Segmented(
            options = ThemeMode.entries.map { it.label },
            active = themeMode.label,
            onSelect = { label ->
                ThemeMode.entries.firstOrNull { it.label == label }?.let(viewModel::setThemeMode)
            },
        )
        Spacer(Modifier.height(16.dp))

        SectionHeader("Screen display mode")
        Text(
            "How paywalls are presented on screen",
            color = tokens.textSec,
            fontSize = 12.sp,
            modifier = Modifier.padding(horizontal = 20.dp),
        )
        Spacer(Modifier.height(8.dp))
        Segmented(
            options = DisplayMode.entries.map { it.label },
            active = displayMode.label,
            onSelect = { label ->
                DisplayMode.entries.firstOrNull { it.label == label }?.let(viewModel::setDisplayMode)
            },
        )
        Spacer(Modifier.height(16.dp))

        SectionHeader("About")
        SettingsCard {
            RowBetween {
                Text("Version", color = tokens.text, fontSize = 15.sp)
                Text("1.0.0", color = tokens.textSec, fontSize = 15.sp)
            }
            Divider()
            RowBetween {
                Text("Purchasely SDK", color = tokens.text, fontSize = 15.sp)
                Text(viewModel.sdkVersion, color = tokens.textSec, fontSize = 15.sp)
            }
        }
        Text(
            "Powered by Purchasely",
            color = tokens.textTer,
            fontSize = 12.sp,
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp, vertical = 16.dp),
        )
        Spacer(Modifier.height(60.dp))
    }
}

@Composable
private fun SectionHeader(title: String) {
    val tokens = Shaker.tokens
    Text(
        title,
        color = tokens.indigoText,
        fontSize = 16.sp,
        fontWeight = FontWeight.SemiBold,
        modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp),
    )
    Spacer(Modifier.height(8.dp))
}

@Composable
private fun SettingsCard(content: @Composable () -> Unit) {
    val tokens = Shaker.tokens
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(tokens.bgCard)
            .border(1.dp, tokens.hair, RoundedCornerShape(16.dp)),
    ) { content() }
    Spacer(Modifier.height(16.dp))
}

@Composable
private fun Divider() {
    Box(Modifier.fillMaxWidth().height(1.dp).background(Shaker.tokens.hair))
}

@Composable
private fun RowBetween(content: @Composable () -> Unit) {
    Row(
        modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        content()
        Spacer(Modifier.weight(1f))
    }
}

@Composable
private fun ToggleRow(
    title: String,
    subtitle: String,
    on: Boolean,
    last: Boolean = false,
    onToggle: (Boolean) -> Unit,
) {
    val tokens = Shaker.tokens
    Row(
        modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(Modifier.weight(1f)) {
            Text(title, color = tokens.text, fontSize = 15.sp, fontWeight = FontWeight.Medium)
            Text(subtitle, color = tokens.textSec, fontSize = 13.sp)
        }
        ShakerToggle(on = on, onToggle = { onToggle(!on) })
    }
}

@Composable
private fun ShakerToggle(on: Boolean, onToggle: () -> Unit) {
    val tokens = Shaker.tokens
    Box(
        modifier = Modifier
            .size(width = 51.dp, height = 31.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(if (on) tokens.indigo else if (tokens.dark) Color(0xFF3D3F54) else Color(0xFFE0DEE9))
            .clickable(onClick = onToggle),
        contentAlignment = Alignment.CenterStart,
    ) {
        Box(
            modifier = Modifier
                .padding(start = if (on) 22.dp else 2.dp)
                .size(27.dp)
                .clip(CircleShape)
                .background(Color.White),
        )
    }
}

@Composable
private fun Segmented(options: List<String>, active: String, onSelect: (String) -> Unit) {
    val tokens = Shaker.tokens
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(tokens.inputBg)
            .padding(4.dp),
        horizontalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        options.forEach { o ->
            val selected = o == active
            Box(
                modifier = Modifier
                    .weight(1f)
                    .height(36.dp)
                    .clip(RoundedCornerShape(10.dp))
                    .background(if (selected) tokens.indigoSoft else Color.Transparent)
                    .clickable { onSelect(o) },
                contentAlignment = Alignment.Center,
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    if (selected) {
                        Icon(Icons.Default.Check, null, tint = tokens.indigoText, modifier = Modifier.size(14.dp))
                        Spacer(Modifier.size(6.dp))
                    }
                    Text(
                        o,
                        color = if (selected) tokens.indigoText else tokens.textSec,
                        fontSize = 13.sp,
                        fontWeight = if (selected) FontWeight.SemiBold else FontWeight.Medium,
                    )
                }
            }
        }
    }
}

@Composable
private fun OutlineButton(label: String, onClick: () -> Unit) {
    val tokens = Shaker.tokens
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(52.dp)
            .clip(RoundedCornerShape(100.dp))
            .border(1.5.dp, tokens.indigoText, RoundedCornerShape(100.dp))
            .clickable(onClick = onClick),
        contentAlignment = Alignment.Center,
    ) {
        Text(label, color = tokens.indigoText, fontSize = 16.sp, fontWeight = FontWeight.SemiBold)
    }
}

File: ios/Shaker/Screens/Settings/SettingsScreen.swift
import SwiftUI

struct SettingsScreen: View {

    @StateObject private var viewModel = SettingsViewModel()
    @EnvironmentObject private var premiumManager: PremiumManager
    @EnvironmentObject private var appViewModel: AppViewModel
    @Environment(\.shakerTokens) private var tokens
    @State private var loginInput = ""
    @State private var hostViewController: UIViewController?

    var body: some View {
        ZStack {
            tokens.bg.ignoresSafeArea()
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    Text("Settings")
                        .font(.system(size: 28, weight: .bold))
                        .foregroundStyle(tokens.text)
                        .padding(.horizontal, 20)
                        .padding(.top, 16)

                    sectionHeader("Account")
                    accountCard

                    sectionHeader("Purchases")
                    VStack(spacing: 10) {
                        outlineButton("Restore purchases") { viewModel.restorePurchases() }
                        outlineButton("Show onboarding") {
                            viewModel.displayOnboardingPaywall(from: hostViewController)
                        }
                    }
                    .padding(.horizontal, 20)

                    sectionHeader("Purchasely SDK")
                    segmented(
                        options: PurchaselySDKMode.allCases.map { $0.title },
                        active: viewModel.sdkMode.title,
                        onSelect: { t in
                            PurchaselySDKMode.allCases.first(where: { $0.title == t }).map { viewModel.setSdkMode($0) }
                        }
                    )
                    Text("Default mode is Paywall Observer — Shaker observes purchases but uses its own paywall UI.")
                        .font(.system(size: 12))
                        .foregroundStyle(tokens.textSec)
                        .padding(.horizontal, 20)

                    sectionHeader("Data privacy")
                    privacyCard
                    Text("Technical processing required for app operation cannot be disabled.")
                        .font(.system(size: 12))
                        .foregroundStyle(tokens.textSec)
                        .padding(.horizontal, 20)

                    sectionHeader("Appearance")
                    segmented(
                        options: ["Light", "Dark", "System"],
                        active: viewModel.themeMode.capitalized,
                        onSelect: { t in
                            viewModel.setThemeMode(t.lowercased())
                        }
                    )

                    sectionHeader("Screen display mode")
                    Text("How paywalls are presented on screen")
                        .font(.system(size: 12))
                        .foregroundStyle(tokens.textSec)
                        .padding(.horizontal, 20)
                    segmented(
                        options: ["Full", "Modal", "Drawer", "Popin"],
                        active: displayLabel(viewModel.displayMode),
                        onSelect: { t in viewModel.setDisplayMode(displayStorage(t)) }
                    )

                    sectionHeader("About")
                    aboutCard

                    Text("Powered by Purchasely")
                        .font(.system(size: 12))
                        .foregroundStyle(tokens.textTer)
                        .frame(maxWidth: .infinity)
                        .padding(.bottom, 60)
                }
            }
        }
        .background { ViewControllerResolver { vc in hostViewController = vc } }
        .navigationBarHidden(true)
        .onAppear { viewModel.prefetchOnboardingPresentation() }
        .alert("Restore", isPresented: Binding(
            get: { viewModel.restoreMessage != nil },
            set: { if !$0 { viewModel.clearRestoreMessage() } }
        )) {
            Button("OK") { viewModel.clearRestoreMessage() }
        } message: { Text(viewModel.restoreMessage ?? "") }
        .alert("SDK Restart Required", isPresented: Binding(
            get: { viewModel.sdkModeRestartMessage != nil },
            set: { if !$0 { viewModel.clearSdkModeRestartMessage() } }
        )) {
            Button("OK") { viewModel.clearSdkModeRestartMessage() }
        } message: { Text(viewModel.sdkModeRestartMessage ?? "") }
    }

    // MARK: - Sections

    private func sectionHeader(_ title: String) -> some View {
        Text(title)
            .font(.system(size: 16, weight: .semibold))
            .foregroundStyle(tokens.indigoText)
            .padding(.horizontal, 20)
    }

    private var accountCard: some View {
        VStack(spacing: 0) {
            if let userId = viewModel.userId {
                HStack(spacing: 12) {
                    ZStack {
                        Circle().fill(tokens.indigo)
                        Text(String(userId.prefix(2)).uppercased())
                            .font(.system(size: 16, weight: .bold))
                            .foregroundStyle(tokens.onIndigo)
                    }
                    .frame(width: 44, height: 44)

                    VStack(alignment: .leading, spacing: 2) {
                        Text("Logged in as")
                            .font(.system(size: 12))
                            .foregroundStyle(tokens.textSec)
                        Text(userId)
                            .font(.system(size: 16, weight: .semibold))
                            .foregroundStyle(tokens.text)
                    }
                    Spacer()
                    Button { viewModel.logout() } label: {
                        HStack(spacing: 6) {
                            Image(systemName: "rectangle.portrait.and.arrow.right")
                                .font(.system(size: 13))
                            Text("Logout").font(.system(size: 14, weight: .semibold))
                        }
                        .foregroundStyle(tokens.danger)
                    }
                }
                .padding(16)
            } else {
                HStack(spacing: 10) {
                    TextField("User ID", text: $loginInput)
                        .textInputAutocapitalization(.never)
                        .autocorrectionDisabled()
                        .font(.system(size: 15))
                        .padding(.horizontal, 14).frame(height: 44)
                        .background(RoundedRectangle(cornerRadius: 12).fill(tokens.bgSubtle))
                        .overlay(RoundedRectangle(cornerRadius: 12).stroke(tokens.hair, lineWidth: 1))

                    Button {
                        viewModel.login(userId: loginInput)
                        loginInput = ""
                    } label: {
                        Text("Login")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundStyle(loginInput.isEmpty ? tokens.textSec : .white)
                            .frame(height: 44).padding(.horizontal, 18)
                            .background(
                                RoundedRectangle(cornerRadius: 12)
                                    .fill(loginInput.isEmpty ? tokens.bgSubtle : tokens.accent)
                            )
                    }
                    .disabled(loginInput.trimmingCharacters(in: .whitespaces).isEmpty)
                }
                .padding(12)
            }
            divider
            HStack {
                Text("Premium status").font(.system(size: 15)).foregroundStyle(tokens.text)
                Spacer()
                Text(premiumManager.isPremium ? "PRO" : "FREE")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundStyle(tokens.indigoText)
                    .padding(.horizontal, 10).padding(.vertical, 3)
                    .background(Capsule().fill(premiumManager.isPremium ? tokens.goldSoft : tokens.indigoSoft))
            }
            .padding(.horizontal, 16).padding(.vertical, 12)
            divider
            HStack {
                VStack(alignment: .leading, spacing: 2) {
                    Text("Anonymous ID").font(.system(size: 13)).foregroundStyle(tokens.textSec)
                    Text(viewModel.anonymousId).font(.system(size: 11, design: .monospaced)).foregroundStyle(tokens.text).lineLimit(1)
                }
                Spacer()
                Button {
                    UIPasteboard.general.string = viewModel.anonymousId
                } label: {
                    Image(systemName: "doc.on.doc")
                        .font(.system(size: 13))
                        .foregroundStyle(tokens.textSec)
                        .frame(width: 32, height: 32)
                        .background(RoundedRectangle(cornerRadius: 8).fill(tokens.inputBg))
                }
            }
            .padding(.horizontal, 16).padding(.vertical, 12)
        }
        .background(RoundedRectangle(cornerRadius: 16).fill(tokens.bgCard))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(tokens.hair, lineWidth: 1))
        .padding(.horizontal, 20)
    }

    private var privacyCard: some View {
        VStack(spacing: 0) {
            toggleRow("Analytics", sub: "Anonymous audience measurement",
                      on: Binding(get: { viewModel.analyticsConsent }, set: { viewModel.setAnalyticsConsent($0) }))
            divider
            toggleRow("Identified analytics", sub: "User-identified analytics",
                      on: Binding(get: { viewModel.identifiedAnalyticsConsent }, set: { viewModel.setIdentifiedAnalyticsConsent($0) }))
            divider
            toggleRow("Personalization", sub: "Personalized content & offers",
                      on: Binding(get: { viewModel.personalizationConsent }, set: { viewModel.setPersonalizationConsent($0) }))
            divider
            toggleRow("Campaigns", sub: "Promotional campaigns",
                      on: Binding(get: { viewModel.campaignsConsent }, set: { viewModel.setCampaignsConsent($0) }))
            divider
            toggleRow("Third-party integrations", sub: "External analytics & integrations",
                      on: Binding(get: { viewModel.thirdPartyConsent }, set: { viewModel.setThirdPartyConsent($0) }))
        }
        .background(RoundedRectangle(cornerRadius: 16).fill(tokens.bgCard))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(tokens.hair, lineWidth: 1))
        .padding(.horizontal, 20)
    }

    private var aboutCard: some View {
        VStack(spacing: 0) {
            HStack {
                Text("Version").font(.system(size: 15)).foregroundStyle(tokens.text)
                Spacer()
                Text(Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "1.0.0")
                    .font(.system(size: 15))
                    .foregroundStyle(tokens.textSec)
            }
            .padding(.horizontal, 16).padding(.vertical, 12)
            divider
            HStack {
                Text("Purchasely SDK").font(.system(size: 15)).foregroundStyle(tokens.text)
                Spacer()
                Text(viewModel.sdkVersion).font(.system(size: 15)).foregroundStyle(tokens.textSec)
            }
            .padding(.horizontal, 16).padding(.vertical, 12)
        }
        .background(RoundedRectangle(cornerRadius: 16).fill(tokens.bgCard))
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(tokens.hair, lineWidth: 1))
        .padding(.horizontal, 20)
    }

    private func toggleRow(_ title: String, sub: String, on: Binding<Bool>) -> some View {
        HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text(title).font(.system(size: 15, weight: .medium)).foregroundStyle(tokens.text)
                Text(sub).font(.system(size: 13)).foregroundStyle(tokens.textSec)
            }
            Spacer()
            Toggle("", isOn: on).labelsHidden().tint(tokens.indigo)
        }
        .padding(.horizontal, 16).padding(.vertical, 12)
    }

    private var divider: some View {
        Rectangle().fill(tokens.hair).frame(height: 1)
    }

    private func outlineButton(_ label: String, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            Text(label)
                .font(.system(size: 16, weight: .semibold))
                .foregroundStyle(tokens.indigoText)
                .frame(maxWidth: .infinity, minHeight: 52)
                .overlay(Capsule().stroke(tokens.indigoText, lineWidth: 1.5))
        }
    }

    private func segmented(options: [String], active: String, onSelect: @escaping (String) -> Void) -> some View {
        HStack(spacing: 4) {
            ForEach(options, id: \.self) { o in
                let selected = o == active
                Button {
                    onSelect(o)
                } label: {
                    HStack(spacing: 6) {
                        if selected {
                            Image(systemName: "checkmark")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundStyle(tokens.indigoText)
                        }
                        Text(o)
                            .font(.system(size: 13, weight: selected ? .semibold : .medium))
                            .foregroundStyle(selected ? tokens.indigoText : tokens.textSec)
                    }
                    .frame(maxWidth: .infinity, minHeight: 36)
                    .background(
                        RoundedRectangle(cornerRadius: 10)
                            .fill(selected ? tokens.indigoSoft : Color.clear)
                    )
                }
            }
        }
        .padding(4)
        .background(RoundedRectangle(cornerRadius: 12).fill(tokens.inputBg))
        .padding(.horizontal, 20)
    }

    private func displayLabel(_ v: String) -> String {
        switch v {
        case "fullscreen": return "Full"
        case "modal": return "Modal"
        case "drawer": return "Drawer"
        case "popin": return "Popin"
        default: return "Full"
        }
    }

    private func displayStorage(_ l: String) -> String {
        switch l {
        case "Full": return "fullscreen"
        case "Modal": return "modal"
        case "Drawer": return "drawer"
        case "Popin": return "popin"
        default: return "fullscreen"
        }
    }
}
